### PR TITLE
feat: Calorimeter hit reco simplification

### PR DESCRIPTION
**Pull request by wdconinc**
### Briefly, what does this PR introduce?
This applies the #666 treatment to the CalorimeterHitReco algorithms (on top of #794):
- put all configuration in a single configuration struct for use with WithPodConfig mixin,
- create a templated factory-factory to use in the plugins,
- replace all old explicit CalorimeterHit factories with the new factory-factory.

### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [x] New feature (issue #__)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [ ] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
No.

### Does this PR change default behavior?
No.

### CODE REVIEW

**Comment by wdconinc** [comment]
Only collection ID differences.

File: src/algorithms/calorimetry/CalorimeterHitRecoConfig.h [hunk 1]
```diff
@@ -0,0 +1,42 @@
+// Subject to the terms in the LICENSE file found in the top-level directory.
+//
```
Comment (veprbl) [inline src/algorithms/calorimetry/CalorimeterHitRecoConfig.h:2]:
```suggestion
// SPDX-License-Identifier: LGPL-3.0-or-later
// Copyright (C) 2022 Wouter Deconinck
```

File: src/factories/calorimetry/CalorimeterHitReco_factoryT.h [hunk 2]
```diff
@@ -0,0 +1,89 @@
+// Copyright 2023, Wouter Deconinck
+// Subject to the terms in the LICENSE file found in the top-level directory.
```
Comment (veprbl) [inline src/factories/calorimetry/CalorimeterHitReco_factoryT.h:2]:
```suggestion
// SPDX-License-Identifier: LGPL-3.0-or-later
// Copyright (C) 2022 Wouter Deconinck
```

File: src/factories/calorimetry/CalorimeterHitReco_factoryT.h [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+//
```
Comment (veprbl) [inline src/factories/calorimetry/CalorimeterHitReco_factoryT.h:3]:
```suggestion
```

**Review by veprbl — APPROVED** [review]
Checked difference between CI artifacts for EDM4eic output and flags. Looks good.

**Pull request merged**